sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -268,7 +270,8 @@ void Scheduler::lockSharedTimedMutex(
   // TODO: after switching to C++17, change std::shared_timed_mutex to
   // std::shared_mutex and use std::lock_guard here both for Windows and Linux.
   while (!Lock.owns_lock()) {
-    Lock.try_lock();
+    Lock.try_lock_for(std::chrono::milliseconds(10));
+    std::this_thread::yield();
```

Comment:
@vladimirlaz, could you please explain why ```yield``` is needed here?

Reply:
without yield while loop acts like endless while loop and occupies the whole CPU when multiple command groups are created in multiple host threads (see PR description)

Reply:
Guess what? Instead of answering in the PR, add a comment in the code! :-)
It will help all the readers, not only the reviewers...

Reply:
done